geoh5py/shared/merging/drape_model.py

```diff
@@ -0,0 +1,171 @@
+                prisms.append(ghost_prism)
+                layers.append(ghost_layer)
+
+                # create the first ghost point
+                ghost_prism, ghost_layer = cls._ghost_point(
+                    temp_prisms[0],
+                    temp_prisms[1],
+                    previous_prism - 1,
+                    previous_layer - 1,
+                )
+
+                # append first ghost prism and layer
+                prisms.append(ghost_prism)
+                layers.append(ghost_layer)
+
+            # add the entity prisms and layers to the list
+            temp_prisms[:, -2] += previous_prism
+            temp_layers[:, 0] += previous_layer
+
+            previous_prism = temp_prisms[-1, -1] + temp_prisms[-1, -2] + 2
+            previous_layer = temp_layers[-1, 0] + 3
+
+            prisms.append(temp_prisms)
+            layers.append(temp_layers)
+
+            # create a last ghost prism and layer
+            ghost_prism, ghost_layer = cls._ghost_point(
+                temp_prisms[-1], temp_prisms[-2], previous_prism - 2, previous_layer - 2
+            )
+
+        # create an object
+        output = cls._type.create(  # type: ignore
+            workspace, prisms=np.vstack(prisms), layers=np.vstack(layers), **kwargs
+        )
+
+        return output
+
+    @staticmethod
+    def _ghost_point(
+        point: np.ndarray,
+        mirror: np.ndarray,
+        previous_prism: int,
+        previous_layer: int,
+    ) -> np.ndarray | np.ndarray:
+        """
+        Create a ghost point prism and layer based on two points.
+        :param point: The point to create the ghost from.
+        :param mirror: The mirror point to create the ghost from.
+        :param previous_prism: The ID of the previous prism.
+        :param previous_layer: The ID of the previous layer.
+        :return: A prism and a layer for the ghost point.
+        """
+
+        # create a mirrored point for last layer
+        ghost_prism: np.ndarray = point.copy()
+        ghost_layer: np.ndarray = np.empty(3)
+        ghost_prism[:3] = 2 * point[:3] - mirror[:3]
+
+        ghost_prism[3] = previous_prism
+        ghost_prism[4] = 1
+
+        ghost_layer[0] = previous_layer
+        ghost_layer[1] = 0
+        ghost_layer[2] = ghost_prism[2]
+
+        return np.expand_dims(ghost_prism, 0), np.expand_dims(ghost_layer, 0)
+
+    @classmethod
+    def merge_data(
+        cls,
+        out_entity,
+        input_entities: list[ObjectBase],
+    ):
+        super().merge_data(out_entity, input_entities)
+
+        ind_map: list = []
+        data_count: int = 0
+        n_values: int = out_entity.n_cells - (len(input_entities) - 1) * 2
+        for ghost, input_entity in enumerate(input_entities):
+            if input_entity.n_cells is None:
+                continue
+            ind_map += [
+                np.arange(data_count, data_count + input_entity.n_cells),
+                [n_values + ghost * 2, n_values + ghost * 2],
```

Comment:
Same value twice? Aren't we missing a +1 somewhere? I think it will end up being the same (all ndv anyway), but not really the parsing intended.

Reply:
Very efficient - love it.

Reply:
you are right